File: src/types/skill.ts
// Marketplace types (from skills.sh)
export interface SkillsSearchResult {
  skills: MarketplaceSkillSummary[];
  count: number;
}

export interface MarketplaceSkillSummary {
  id: string;
  name: string;
  source: string;
  skillId: string;
  installs: number;
  installed: boolean;
}

export interface MarketplaceSkillDetail {
  id: string;
  name: string;
  source: string;
  skillId: string;
  installs: number;
  description: string;
  content: string;
}

// Installed skill (persisted state — content excluded for list payloads)
export interface InstalledSkill {
  id: string;
  name: string;
  skillId: string;
  source: string;
  description: string;
  enabled: boolean;
  installs?: number;
}

// Skill content response (for detail view)
export interface SkillContentResponse {
  id: string;
  name: string;
  content: string;
}

// Skill tool info (for Settings > Skills)
export interface SkillToolInfo {
  id: string;
  name: string;
  installed: boolean;
  enabled: boolean;
  skillsPath: string;
}

File: src-tauri/src/state/skill.rs
use serde::{Deserialize, Serialize};

/// An installed skill, persisted in the store and synced to AI tool directories.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct InstalledSkill {
    /// Full skill path, e.g. "vercel-labs/agent-skills/vercel-react-best-practices"
    pub id: String,
    /// Display name from SKILL.md frontmatter
    pub name: String,
    /// Directory name, e.g. "vercel-react-best-practices"
    pub skill_id: String,
    /// GitHub repo, e.g. "vercel-labs/agent-skills"
    pub source: String,
    /// Description from SKILL.md frontmatter
    pub description: String,
    /// The full SKILL.md content (stored for offline use)
    pub content: String,
    /// Whether this skill is currently active
    pub enabled: bool,
    /// Install count from marketplace at install time
    pub installs: Option<u64>,
}
